### PR TITLE
Merge master -> 1.0.x

### PR DESCRIPTION
**Pull request by mwaggett**
I'm hoping this PLATFORM_INFO reorg will be the last one we need to ensure smooth merge-ups in the future.

CHANGELOG conflicts resolved by adding info from master CHANGELOG to 1.0.x one.

lib/packaging/platforms.rb conflicts resolved by realigning everything to look like master.

### CODE REVIEW

**Comment by mwaggett** [comment]
I know this looks like a lot, but if you look at https://github.com/puppetlabs/packaging/pull/824/files?w=1, you can see that most of it is whitespace changes.

File: CHANGELOG.md [hunk 1]
```diff
@@ -126,7 +136,9 @@ This changelog adheres to [Keep a CHANGELOG](http://keepachangelog.com/).
 [0.99.3]: https://github.com/puppetlabs/packaging/compare/0.99.2...0.99.3
 [0.99.2]: https://github.com/puppetlabs/packaging/compare/0.99.1...0.99.2
 [0.99.1]: https://github.com/puppetlabs/packaging/compare/0.99.0...0.99.1
-[0.99.0]: https://github.com/puppetlabs/packaging/compare/0.6.2...0.99.0
+[0.99.0]: https://github.com/puppetlabs/packaging/compare/0.6.4...0.99.0
```
Comment (underscorgan) [inline CHANGELOG.md]: This should stay at 0.6.2...0.99.0 since that's the actual changeset at the time of the release

Comment (mwaggett) [reply CHANGELOG.md]: ahh okay!

File: CHANGELOG.md [hunk 2]
```diff
@@ -127,6 +137,8 @@ This changelog adheres to [Keep a CHANGELOG](http://keepachangelog.com/).
 [0.99.2]: https://github.com/puppetlabs/packaging/compare/0.99.1...0.99.2
 [0.99.1]: https://github.com/puppetlabs/packaging/compare/0.99.0...0.99.1
 [0.99.0]: https://github.com/puppetlabs/packaging/compare/0.6.2...0.99.0
```
Comment (mwaggett) [inline CHANGELOG.md]: @underscorgan fixed!

**Pull request merged**